edk2toolext/capsule/capsule_helper.py

```diff
@@ -0,0 +1,185 @@
+## @file capsule_helper.py
+# This module contains helper functions for building EDK2 FMP UEFI capsules from
+# binary payloads, along with the functions to standardize the creation of the Windows
+# driver installation files.
+#
+##
+# Copyright (C) Microsoft Corporation
+#
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+
+import uuid
+import os
+import struct
+
+from edk2toollib.windows.capsule import inf_generator, cat_generator
+from edk2toollib.uefi.uefi_capsule_header import UefiCapsuleHeaderClass
+from edk2toollib.uefi.fmp_capsule_header import FmpCapsuleHeaderClass, FmpCapsuleImageHeaderClass
+from edk2toollib.uefi.fmp_auth_header import FmpAuthHeaderClass
+from edk2toollib.uefi.edk2.fmp_payload_header import FmpPayloadHeaderClass
+
+# https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.pkcs.contentinfo.-ctor?view=netframework-4.8
+PKCS7_SIGNED_DATA_OID = '1.2.840.113549.1.7.2'
+
+
+def get_capsule_file_name(capsule_options):
+    '''from the shared capsule_options dictionary, returns the formatted capsule file name'''
+    return f"{capsule_options['fw_name']}_{capsule_options['fw_version_string']}.bin"
+
+
+def get_normalized_version_string(version_string):
```

Comment:
not critical...but seems like this really belongs in cat gen tool rather than here since this code should be somewhat agnostic from windows capsules

Reply:
I tend to agree. To get the code to land (since the InfGenerator is in library, not extensions), I'll keep it here for now, but will open an issue to move it.